app/src/main/java/com/codepatissier/keki/src/main/consumer/search/ConsumerSearchFragment.kt

```diff
@@ -1,54 +1,60 @@
 import com.codepatissier.keki.databinding.FragmentConsumerSearchBinding
-import com.codepatissier.keki.src.main.consumer.search.model.PopularSearchesResponse
+import com.codepatissier.keki.src.main.consumer.search.model.MainSearchesResponse
+import com.codepatissier.keki.src.main.consumer.search.model.PatchSearchResponse
+import com.codepatissier.keki.src.main.consumer.search.searchresult.ConsumerSearchActivity
 import com.codepatissier.keki.util.recycler.search.*
 
-class ConsumerSearchFragment : BaseFragment<FragmentConsumerSearchBinding>(FragmentConsumerSearchBinding::bind, R.layout.fragment_consumer_search) , SearchView{
+class ConsumerSearchFragment : BaseFragment<FragmentConsumerSearchBinding>(FragmentConsumerSearchBinding::bind, R.layout.fragment_consumer_search) , SearchMainView{
 
-    lateinit var searchRecentAdapter : SearchRecentAdapter
-    val searchRecentData = mutableListOf<SearchRecentData>()
-
-    lateinit var searchPopularAdapter : SearchPopularAdapter
-    val searchPopularData = mutableListOf<SearchPopularData>()
-
-    lateinit var searchRecentCakeImgAdapter: SearchCakeImgAdapter
-    val searchCakeImgData = mutableListOf<SearchCakeImgData>()
+    private lateinit var searchRecentAdapter : SearchRecentAdapter
+    private lateinit var searchPopularAdapter : SearchPopularAdapter
+    private lateinit var searchRecentPostAdapter: SearchRecentPostAdapter
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-
-
-        deleteSearchHistory()
-        searchRecentRecycler()
-        searchRecentSeenRecycler()
         setListenerToEditText()
-
         showLoadingDialog(requireContext())
-        SearchService(this).tryGetPopularSearches()
+        SearchMainService(this).tryGetMainSearches()
+        deleteSearchHistory()
     }
 
-    override fun onGetPopularSearchesSuccess(response: PopularSearchesResponse) {
+    override fun onGetMainSearchesSuccess(response: MainSearchesResponse) {
         dismissLoadingDialog()
-        searchPopularRecycler(response)
+        searchMainRecycler(response)
     }
 
-    override fun onGetPopularSearchesFailure(message: String) {
+    override fun onGetMainSearchesFailure(message: String) {
         dismissLoadingDialog()
         showCustomToast("오류 : $message")
     }
 
-    //검색어 전체 지우기 클릭 이벤트
+    override fun onPatchSearchesSuccess(response: PatchSearchResponse) {
+        deleteSearchHistory()
+    }
+
+    override fun onPatchSearchesFailure(message: String) {
+        showCustomToast("삭제 오류 : $message")
+    }
+
+    override fun onResume() {
+        super.onResume()
+        SearchMainService(this).tryGetMainSearches()
+    }
+
+    //검색어 전체 지우기 -> 최근 검색어 뷰 없어짐
     private fun deleteSearchHistory(){
         binding.tvRecentSearchDelete.setOnClickListener {
+            SearchMainService(this).tryPatchSearchHistories()
             binding.llEmptyHistory.visibility = View.GONE
         }
     }
```

Comment:
onPatchSearchesSuccess() 함수에 deleteSearchHistory() 말고 binding.llEmptyHistory.visibility = View.GONE을 쓰는 게 더 맞는 거 같아요! (deleteSearchHistory()에서는 visibility 코드는 삭제하고)
=> 최근검색어 지우기를 클릭했을 땐 서버에 request를 보내고,
그 요청이 성공했을 때 visibility를 GONE 처리하기 

그리고 onResume()에서 한번 더 GET method를 호출하는 이유가 궁금해요!

```suggestion
    override fun onPatchSearchesSuccess(response: PatchSearchResponse) {
        binding.llEmptyHistory.visibility = View.GONE
    }
    
    override fun onPatchSearchesFailure(message: String) {
        showCustomToast("삭제 오류 : $message")
    }
    
    private fun deleteSearchHistory(){
        binding.tvRecentSearchDelete.setOnClickListener {
            SearchMainService(this).tryPatchSearchHistories()
        }
    }
```

Reply:
확인했습니다! 수고했어용 :thumbsup: